scvi/train/_trainingplans.py

```diff
@@ -298,14 +301,19 @@ def compute_and_log_metrics(
             Postfix string to add to the metric name of
             extra metrics
         """
-        rec_loss = loss_recorder.reconstruction_loss
-        n_obs_minibatch = rec_loss.shape[0]
-        rec_loss = rec_loss.sum()
-        kl_local = loss_recorder.kl_local.sum()
-        kl_global = loss_recorder.kl_global
-
-        # use the torchmetric object for the ELBO
-        metrics.update(
+        if isinstance(loss_recorder, LossRecorder):
+            loss_output = loss_recorder._loss_output
+        else:
+            loss_output = loss_recorder
+        rec_loss = loss_output.reconstruction_loss_sum
+        n_obs_minibatch = loss_output.n_obs_minibatch
+        kl_local = loss_output.kl_local_sum
+        kl_global = loss_output.kl_global_sum
+
+        # Use the torchmetric object for the ELBO
+        # We only need to update the ELBO metric
+        # As it's defined as a sum of the other metrics
+        next(iter(metrics.values()))[0].update(
```

Comment:
why can't you key into elbo here?

Reply:
ah yeah I didn't see we have access to the mode here